src/main/java/uk/gov/hmcts/reform/translate/service/DictionaryService.java

```diff
@@ -103,23 +107,27 @@ public void putDictionary(final Dictionary dictionaryRequest) {
         validateDictionary(dictionaryRequest, isManageTranslationRole);
         val currentUserId = securityUtils.getUserInfo().getUid();
 
-        val translationUploadOptional = hasAnyTranslations(dictionaryRequest)
+        val translationUploadEntity = hasAnyTranslations(dictionaryRequest)
             ? dictionaryMapper.createTranslationUploadEntity(currentUserId)
             : null;
 
+        // if upload entity has been generated save it now as we know we have at least one translation that will use it
+        if (translationUploadEntity != null) {
+            translationUploadRepository.save(translationUploadEntity);
+        }
+
```

Comment:
No sure if we need to create and save at this point as it is only needed before update.

Reply:
resolved in 35490b1863657a77ef2e79dcba826e9dde1d0f4d